include/boost/fusion/support/detail/and.hpp

```diff
@@ -0,0 +1,35 @@
+/*=============================================================================
+    Copyright (c) 2016 Lee Clagett
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+==============================================================================*/
+#ifndef FUSION_AND_07152016_1625
+#define FUSION_AND_07152016_1625
+
+#include <boost/config.hpp>
+#include <boost/type_traits/integral_constant.hpp>
+
+#if defined(BOOST_NO_CXX11_VARIADIC_TEMPLATES)
+#error fusion::detail::and_ requires variadic templates
+#endif
+
+namespace boost { namespace fusion { namespace detail {
+    template<typename ...Cond>
+    struct and_impl : false_type {};
+
+    template<typename ...T>
+    struct and_impl<true_type, integral_constant<T, true>...> : true_type {};
+   
+    /* fusion::detail::and_ differs from mpl::and_ in the following ways:
+       - The empty set is valid and returns false
+       - A single element set is valid and returns the identity
+       - There is no upper bound on the set size
+       - The conditions are evaluated at once, and are not short-circuited. This
+         reduces instantations when returning true; the implementation is not
+         recursive. */
+    template<typename ...Cond>
+    struct and_ : and_impl<integral_constant<bool, Cond::value>...> {};
```

Comment:
The first `true_type` in the specialization prevents it from matching the empty-set case. Returning true on the empty-set might make more sense. I chose false because `std::tuple` disables the variadic constructions if zero elements are are provided, so this implementation automatically disables the constructor without an additional check. Perhaps that check should be explicit?